price_feed/lib.rs

```diff
@@ -0,0 +1,42 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[ink::contract]
+mod price_feed {
+    use ink::prelude::vec::Vec;
+    use runtime_common::chain_ext::{CoinInfo, Blockchain, Symbol, ChainExtensionError};
+
+    #[ink(storage)]
+    pub struct PriceFeed {}
+    impl PriceFeed {
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self {}
+        }
+        #[ink(message)]
+        pub fn get_coin_info(
+            &mut self,
+            blockchain: Blockchain,
+            symbol: Symbol,
+        ) -> Result<CoinInfo, ChainExtError> {
+            let input = (blockchain, symbol);
+            ::ink::env::chain_extension::ChainExtensionMethod::build(7777u32)
+                .input::<(Blockchain, Symbol)>()
+                .output::<Result<CoinInfo, ChainExtError>, false>()
+                .handle_error_code::<ChainExtError>()
+                .call(&input)?
+        }
+    }
+
+    #[derive(Debug, Copy, Clone, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub struct ChainExtError(ChainExtensionError);
```

Comment:
The naming here could be a bit confusing but I can't think of something better. Maybe import the `ChainExtensionError` as `RuntimeChainExtensionError` and rename `ChainExtError` to `ChainExtensionError`?
Not that important though